realm/realm-library/src/main/java/io/realm/HandlerController.java

```diff
@@ -450,6 +461,16 @@ private void completedAsyncQueriesUpdate(QueryUpdateTask.Result result) {
                     asyncRealmResults.remove(weakRealmResults);
 
                 } else {
+                    // update the instance with the new pointer
+                    try {
+                        realmResults.swapTableViewPointer(query.getValue());
+                    } catch (BadVersionException e) {
```

Comment:
same as above. we're either on the same version, or we just advanced to the worker thread version, which should crash here https://github.com/realm/realm-java/pull/2450/files#diff-c6f79e9457082a73f91697a0d6ed683dR451 before we can get to this block